packages/shared/src/utils/deprecated.test.ts

```diff
@@ -0,0 +1,201 @@
+jest.mock('./runtimeEnvironment', () => {
+  return { isDevelopmentEnvironment: jest.fn(() => true) };
+});
+
+import { deprecated, deprecatedProperty } from './deprecated';
+import { isDevelopmentEnvironment } from './runtimeEnvironment';
+
+describe('deprecated(fnName, warning)', () => {
+  let consoleWarnSpy;
+
+  beforeEach(() => {
+    consoleWarnSpy = jest.spyOn(global.console, 'warn').mockImplementation();
+  });
+  afterEach(() => {
+    consoleWarnSpy.mockRestore();
+  });
+
+  test('deprecate class method shows warning', () => {
+    class Example {
+      getSomeMethod = (arg1?, arg2?) => {
+        deprecated('getSomeMethod', 'Use `getSomeMethodElse` instead.');
+        return `getSomeMethodValue:${arg1 || '-'}:${arg2 || '-'}`;
+      };
+    }
+
+    const example = new Example();
+
+    expect(consoleWarnSpy).not.toBeCalled();
+    expect(example.getSomeMethod('a', 'b')).toEqual('getSomeMethodValue:a:b');
+    expect(consoleWarnSpy).toBeCalledTimes(1);
+    expect(consoleWarnSpy).toBeCalledWith(
+      'DEPRECATION WARNING: "getSomeMethod" is deprecated and will be removed in the next major release.\nUse `getSomeMethodElse` instead.',
+    );
+
+    expect(example.getSomeMethod()).toEqual('getSomeMethodValue:-:-');
+    expect(consoleWarnSpy).toBeCalledTimes(1);
+  });
+
+  test('deprecate static method shows warning', () => {
+    class Example {
+      static getSomeStaticMethod = () => {
+        deprecated('getSomeStaticMethod', 'Use `getSomeStaticMethodElse` instead.');
+        return 'getSomeStaticMethodValue';
+      };
+    }
+
+    expect(consoleWarnSpy).not.toBeCalled();
+    expect(Example.getSomeStaticMethod()).toEqual('getSomeStaticMethodValue');
+    expect(consoleWarnSpy).toBeCalledTimes(1);
+    expect(consoleWarnSpy).toBeCalledWith(
+      'DEPRECATION WARNING: "getSomeStaticMethod" is deprecated and will be removed in the next major release.\nUse `getSomeStaticMethodElse` instead.',
+    );
+
+    expect(Example.getSomeStaticMethod()).toEqual('getSomeStaticMethodValue');
+    expect(consoleWarnSpy).toBeCalledTimes(1);
+  });
+
+  test('deprecate function shows warning', () => {
+    const getSomeFunction = () => {
+      deprecated('getSomeFunction', 'Use `getSomeFunctionElse` instead.');
+      return 'getSomeFunctionValue';
+    };
+
+    expect(consoleWarnSpy).not.toBeCalled();
+    expect(getSomeFunction()).toEqual('getSomeFunctionValue');
+    expect(consoleWarnSpy).toBeCalledTimes(1);
+    expect(consoleWarnSpy).toBeCalledWith(
+      'DEPRECATION WARNING: "getSomeFunction" is deprecated and will be removed in the next major release.\nUse `getSomeFunctionElse` instead.',
+    );
+
+    expect(getSomeFunction()).toEqual('getSomeFunctionValue');
+    expect(consoleWarnSpy).toBeCalledTimes(1);
+  });
```

Comment:
Last nit:

Copy/paste that test and use a `key` for the deprecated function. Then no key + key usage is tested

Reply:
I've introduced the following 2 tests:
- deprecate function with key shows warning
- deprecate function with the same key does not show warning